src/components/widgets-settings/formula/formula.directive.coffee

```diff
@@ -48,15 +48,18 @@ module.controller('SettingFormulaCtrl', ($scope, $filter, $timeout) ->
       i++
     )
 
-    # Guard against injection
-    if (!str.match(authorized_regex))
-      w.isFormulaCorrect = false
-      w.evaluatedFormula = "invalid expression"
+    w.evaluatedFormula = applyFormula(str)
 
-    try
-      w.evaluatedFormula = eval(str).toFixed(2)
-    catch e
-      w.evaluatedFormula = "invalid expression"
+    if w.selectedAccounts? && firstAcc = w.selectedAccounts[0]
```

Comment:
@an-pavlovych code duplication here. The string replacement could have been put in a function and re-used